### PR TITLE
feat: add BaseMail class

### PR DESCRIPTION
**Pull request by Goodmain**

### CODE REVIEW

File: src/Mail/BaseMail.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    use SerializesModels;
+
+    public int $tries = 5;
+
+    public function __construct(array $viewData, $view, $subject)
+    {
+        $this->viewData = $viewData;
+        $this->subject = $subject;
+        $this->view = $view;
+        $this->onQueue('mails');
```
Comment (DenTray) [inline src/Mail/BaseMail.php:22]:
```suggestion
    public int $tries = 5;
    public string $queue = 'mails';

    public function __construct(array $viewData, $view, $subject)
    {
        $this->viewData = $viewData;
        $this->subject = $subject;
        $this->view = $view;
        $this->onQueue($this->queue);
```

**Review by DenTray — APPROVED** [review]

File: src/Mail/BaseMail.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    public int $tries = 5;
+    public string $queue = 'mails';
+
+    public function __construct(array $viewData, $view, $subject)
+    {
+        $this->viewData = $viewData;
+        $this->subject = $subject;
+        $this->view = $view;
+        $this->onQueue($this->queue);
```
Comment (DenTray) [inline src/Mail/BaseMail.php:23]:
```suggestion
        $this->onQueue($this->queue ?? 'mails');
```

File: src/Mail/BaseMail.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    /**
+     * @deprecated
+     * @codeCoverageIgnore
+     */
+    public function build()
+    {
+        return $this
+            ->view($this->view, $this->viewData)
+            ->subject($this->subject)
+            ->onQueue('mails');
```
Comment (DenTray) [inline src/Mail/BaseMail.php:35]:
```suggestion
            ->onQueue($this->queue ?? 'mails');
```

File: src/Mail/BaseMail.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+use Illuminate\Mail\Mailable;
+use Illuminate\Queue\SerializesModels;
+
+class BaseMail extends Mailable implements ShouldQueue
+{
+    use Queueable;
+    use SerializesModels;
+
+    public int $tries = 5;
+    public string $queue = 'mails';
```
Comment (DenTray) [inline src/Mail/BaseMail.php:16]:
```suggestion
```

**Review by DenTray — CHANGES_REQUESTED** [review]

File: src/Mail/BaseMail.php (hunk 2 above)
Comment (Goodmain) [reply src/Mail/BaseMail.php]: I made it simplier. I decide we simply could override the queue if it necessary.

```php
class BaseMail extends Mailable implements ShouldQueue
{
    use Queueable;
    use SerializesModels;

    public int $tries = 5;

    public function __construct(array $viewData, $subject, $view)
    {
        $this->viewData = $viewData;
        $this->subject = $subject;
        $this->view = $view;
        $this->queue = 'mails';
    }
}
```

```php
class SomeMail extends BaseMail
{
    public function __construct(array $viewData)
    {
       parent::__construct(
            $viewData,
            'Subject of this mail',
            'emails.my_new_template'
        );

        $this->queue = 'different_queue';
    }
}
```

**Review by DenTray — APPROVED** [review]

**Review by DenTray — APPROVED** [review]

**Pull request merged**